renderdoc/driver/gl/wrappers/gl_emulated.cpp

```diff
@@ -3036,19 +3047,35 @@ void APIENTRY _glGetTexImage(GLenum target, GLint level, const GLenum format, co
 
     GL.glUseProgram(prog);
 
-    loc = GL.glGetUniformLocation(prog, "srcTex");
-    GL.glUniform1i(loc, 0);
-    loc = GL.glGetUniformLocation(prog, "res");
-    GL.glUniform2f(loc, float(width), float(height));
-
-    GL.glDrawArrays(eGL_TRIANGLES, 0, 3);
+    GLenum depthMode = eGL_DEPTH_COMPONENT;
+    if(depthFormat && HasExt[ARB_stencil_texturing])
+    {
+      GL.glGetTexParameteriv(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, (GLint *)&depthMode);
+    }
 
-    // if we support reading stencil, read the stencil into green
-    if(remapformat == eGL_DEPTH_STENCIL && HasExt[ARB_stencil_texturing])
+    for(int32_t d = 0; d < depth; ++d)
     {
-      GL.glTexParameteri(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, eGL_STENCIL_INDEX);
-      GL.glColorMask(GL_FALSE, GL_TRUE, GL_FALSE, GL_FALSE);
+      GL.glColorMask(GL_TRUE, GL_TRUE, GL_TRUE, GL_TRUE);
+      GL.glTexParameteri(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, eGL_DEPTH_COMPONENT);
```

Comment:
I just realised that this `glTexParameteri` call is now unconditional and doesn't check whether we have a depth format or whether stencil texturing is supported. You should put it inside the same `if(depthFormat && HasExt[ARB_stencil_texturing])` as the other calls.